packages/samples/headless-ssr-commerce/app/_lib/commerce-engine-config.ts

```diff
@@ -57,7 +60,12 @@ export default {
     },
   },
   controllers: {
-    summary: defineQuerySummary(),
-    productList: defineProductList(),
+    summaryListing: defineQuerySummary({listing: true}),
+    summarySearch: defineQuerySummary({search: true}),
+    productListListing: defineProductList({listing: true}),
+    productListSearch: defineProductList({search: true}),
```

Comment:
I don't think you have to do that (i.e., defining every controller twice); if you just do `defineQuerySummary()`, search and listing will be true by default and then the correct controller (or sub-controller) should automatically be used depending on the engine definition you use.

@y-lakhdar correct me if I'm wrong 😁

Reply:
removed them !